src/ophyd_async/epics/adcore/_hdf_writer.py

```diff
@@ -90,16 +97,29 @@ async def open(self, multiplier: int = 1) -> Dict[str, DataKey]:
             )
         ]
         # And all the scalar datasets
-        for ds_name, ds_path in self._scalar_datasets_paths.items():
-            self._datasets.append(
-                HDFDataset(
-                    f"{name}-{ds_name}",
-                    f"/entry/instrument/NDAttributes/{ds_path}",
-                    (),
-                    "",
-                    multiplier,
-                )
-            )
+        for plugin in self._plugins:
+            maybe_xml = await plugin.nd_attributes_file.get_value()
+            # This is the check that ADCore does to see if it is an XML string
+            # rather than a filename to parse
+            if "<Attributes>" in maybe_xml:
+                root = ET.fromstring(maybe_xml)
+                for child in root:
+                    datakey = child.attrib["name"]
+                    self._datasets.append(
+                        HDFDataset(
+                            datakey,
+                            f"/entry/instrument/NDAttributes/{datakey}",
+                            (),
+                            convert_pv_dtype_to_np(
+                                child.attrib.get("dbrtype", "DBR_NATIVE")
+                            )
+                            if child.attrib.get("type", "EPICS_PV") == "EPICS_PV"
+                            else convert_param_dtype_to_np(
+                                child.attrib.get("datatype", "INT")
+                            ),
+                            multiplier,
+                        )
+                    )
```

Comment:
Please can we extract datatype definition
```suggestion
                    if child.attrib.get("type", "EPICS_PV") == "EPICS_PV"
                        np_datatype = convert_pv_dtype_to_np(
                                child.attrib.get("dbrtype", "DBR_NATIVE")
                        )
                    else:
                        np_datatype = convert_param_dtype_to_np(
                                child.attrib.get("datatype", "INT")
                        )
                    self._datasets.append(
                        HDFDataset(
                            datakey,
                            f"/entry/instrument/NDAttributes/{datakey}",
                            (),
                            np_datatype,
                            multiplier,
                        )
                    )
```